File: Selenium/Activities/Activity14.java
package activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;

import java.time.Duration;
import java.util.List;

public class Activity14 {
    public static void main(String[] args) throws InterruptedException {
        //Setup GekoDriver
        WebDriverManager.firefoxdriver().setup();

        //Create driver Object for Firefox Browser
        FirefoxDriver driver = new FirefoxDriver();

        //Set Implicit wait of 5 Secs
        driver.manage().timeouts().implicitlyWait(Duration.ofMillis(5000));
        //Open the url
        driver.get("https://v1.training-support.net/selenium/tables");

        //Get the Title
        String title = driver.getTitle();
        System.out.print("Title of the Page is : " + title + "\n");

        List<WebElement> tableRows = driver.findElements(By.xpath("//table[@id='sortableTable']/tbody/tr"));
        System.out.printf("Number of Records of Sortable Table : %d",  tableRows.size());

        List<WebElement> tableCols = driver.findElements(By.xpath("//table[@id='sortableTable']/tbody/tr[1]/td"));
        System.out.printf("\nNumber of Columns of Sortable Table : %d\n",tableCols.size());

        WebElement cell = driver.findElement(By.xpath("//table[@id='sortableTable']/tbody/tr[2]/td[2]"));
        System.out.println(cell.getText());

        WebElement tableheader = driver.findElement(By.xpath("//table[@id='sortableTable']/thead/tr/th[1]"));
        tableheader.click();

        WebElement cell21 = driver.findElement(By.xpath("//table[@id='sortableTable']/tbody/tr[2]/td[2]"));
        System.out.println(cell21.getText());

        List<WebElement> tablefotter = driver.findElements(By.xpath("//table[@id='sortableTable']/tfoot/tr/th"));
        for (int i=0; i<tablefotter.size(); i++){
            System.out.println(tablefotter.get(i).getText());
        }

        //Wait for 5 Seconds and Close the Browser
        Thread.sleep(5000);
        driver.quit();

    }
}

File: Selenium/Activities/Activity14.py
# Import required packages from selenium
import time

from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.wait import WebDriverWait
from webdriver_manager.firefox import GeckoDriverManager

# Set up the Firefox Driver using WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Set Explicit wait of 3 Secs
    wait = WebDriverWait(driver, 10)

    # Launch the URL and get the title
    driver.get("https://v1.training-support.net/selenium/tables")
    home_title = driver.title
    print("Title of the Home Page is : {}".format(home_title))

    tableRows = driver.find_elements(By.XPATH, "//table[@id='sortableTable']/tbody/tr")
    print("Number of Records of Sortable Table : {}".format(len(tableRows)))

    tableCols = driver.find_elements(By.XPATH, "//table[@id='sortableTable']/tbody/tr[1]/td")
    print("Number of Columns of Sortable Table : {}".format(len(tableCols)))

    cell = driver.find_element(By.XPATH, "//table[@id='sortableTable']/tbody/tr[2]/td[2]")
    print("Text at 2nd row, 2nd column is : {} ".format(cell.text))

    tableheader = driver.find_element(By.XPATH, "//table[@id='sortableTable']/thead/tr/th[1]")
    tableheader.click()

    cell21 = driver.find_element(By.XPATH, "//table[@id='sortableTable']/tbody/tr[2]/td[1]")
    print("Text at 2nd row, 1nd column is : {} ".format(cell21.text))

    tablefotter = driver.find_elements(By.XPATH, "//table[@id='sortableTable']/tfoot/tr/th")
    for footer in tablefotter:
        print(footer.text)

    # Wait for 5 Seconds and Close the Browser
    time.sleep(3)
    driver.close()
